vm/src/serde/deserialize_utils.rs

```diff
@@ -1,3 +1,10 @@
+//! # Deserialization utils
+//!
+//! This module contains some helper functions used in [`Program`](crate::types::program::Program) deserialization.
+//! Namely, [`maybe_add_padding`] and [`take_until_unbalanced`].
```

Comment:
These are specific utils, and not something end users should be importing (as we might deprecate/change them according to our needs). Could we make them (and if we can the whole module) private?

Reply:
Sure! Doing that I also found one of the errors isn't even being used 😅